python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1138 @@
+            elif input_node.node().kind() == "prim::Constant":
+                input_list_r.append(self._consts[input_node.debugName()])
+            else:
+                input_list_r.append("call/var."+input_node.debugName())
+
+                # If the inputs of a ListConstruct op is a call or var, remove it from inputs
+                if op_node.kind() == 'prim::ListConstruct':
+                    if op_name in self._inputs_r.keys():
+                        self._inputs_r.pop(op_name)
+
+            try:
+                input_node_kind = input_node.type().kind()
+                if input_node_kind == 'TensorType':
+                    if input_node.type().scalarType() is None:
+                        input_list_types.append('float')
+                    else:
+                        input_list_types.append(input_node.type().scalarType().lower())
+                elif input_node_kind == 'ListType':
+                    input_list_types.append(str(input_node.type().getElementType()).lower())
+                elif input_node_kind == 'IntType' or input_node_kind == 'FloatType' or \
+                        input_node_kind == 'BoolType' or input_node_kind == 'StringType' or \
+                        input_node_kind == 'OptionalType':
+                    input_list_types.append(str(input_node.type()).lower())
+                else:
+                    input_list_types.append('UnsupportedType')
+            except Exception as e:
+                print('Internal PyTorch error. Failed to grab type.')
+
+        node_str = str(op_node)
+        node_assign = (node_str.split(' = ')[0]).split(' : ')
+        node_type = node_assign[1]
+
+        if op_node.kind() == 'aten::ones':
+            node_type = node_type.split('(')[0]
+            input_list_types[0] = node_type.lower()
+
+        if op_node.kind() == 'aten::zeros':
+            node_type = node_type.split('(')[0]
+            input_list_types[0] = node_type.lower()
+
+        self._op_inputs_r[(op_name, operator)] = input_list_r
+        self._op_inputs_types[(op_name, operator)] = input_list_types
+
+
+    def _parse_import_prerequisites(self):
+        """ Calculate the named preconditions from PyTorch graph.
+
+        Returns
+        -------
+        missing_operators : set object
+            Set of operator names which don't have their mapping in TVM,
+            i.e. which are not supported
+
+        """
+        missing_operators = set()
+        for node in self._trace.graph.nodes():
+            if node.kind() == "prim::Constant" or node.kind() == 'prim::ListConstruct' or \
+                    node.kind() == 'prim::GetAttr':
+                pass
+            else:
+                if any([node.kind() in _convert_map]):
+                    pass
+                else:
+                    missing_operators.add(node.kind())
+
+        return missing_operators
+
+def from_pytorch(trace, input_shapes, input_types):
```

Comment:
why do you need input_types here? Other frontends do not take this

Reply:
You're right, we can grab the input from the PT graph.